castervoice/lib/ccr/core/text_manipulation.py

```diff
@@ -0,0 +1,260 @@
+            R(ContextAction(default=Function(text_manipulation_functions.select_until_phrase, 
+            dict(dictation="phrase", lease_ross="left_right"), 
+            cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_until_phrase, dict(dictation="phrase", lease_ross="left_right"), 
+            cursor_behavior="texstudio"))]),
+                 rdescript="Text Manipulation: select until chosen phrase"),
+        "grab <lease_ross> [<number_of_lines_to_search>] until [<before_after>] [<occurrence_number>] <character>":
+            R(ContextAction(default=Function(text_manipulation_functions.select_until_phrase,
+            dict(character="phrase", lease_ross="left_right"), 
+            cursor_behavior="standard"), actions=[(AppContext("texstudio"),
+            Function(text_manipulation_functions.select_until_phrase, 
+            dict(character="phrase", lease_ross="left_right"), 
+            cursor_behavior="texstudio"))]),
+            rdescript="Text Manipulation: select until chosen character"),
+        "remove <lease_ross> [<number_of_lines_to_search>] until [<before_after>] [<occurrence_number>] <dictation>":
+            R(ContextAction(default=Function(text_manipulation_functions.copypaste_delete_until_phrase,
+                       dict(dictation="phrase", lease_ross="left_right"), cursor_behavior="standard"),
+                       actions=[(AppContext("texstudio"), Function(text_manipulation_functions.copypaste_delete_until_phrase,
+                       dict(dictation="phrase", lease_ross="left_right"), cursor_behavior="texstudio"))]),
+              rdescript="Text Manipulation: delete until chosen phrase"),
+        "remove <lease_ross> [<number_of_lines_to_search>] until [<before_after>] [<occurrence_number>] <character>":
+            R(ContextAction(default=Function(text_manipulation_functions.copypaste_delete_until_phrase,
+                       dict(character="phrase", lease_ross="left_right"),
+                       cursor_behavior="standard"),
+                       actions=[(AppContext("texstudio"), Function(text_manipulation_functions.copypaste_delete_until_phrase,
+                       dict(character="phrase", lease_ross="left_right"),
+                       cursor_behavior="texstudio"))]),
+              rdescript="Text Manipulation: delete until chosen character"),
+        
+
+
+        
+    }
+    character_dict = {
+                "(left prekris | lay)": "(",
+                "(right prekris | ray)": ")",
+                "(left brax | lack)": "[",
+                "(right brax | rack)": "]",
+                "(left angle | lang)": "<",
+                "(right angle | rang)": ">",
+                "(left curly | lace)": "{",
+                "(right curly | race)": "}",
+                "quotes": '"',
+                "(single quote | thin quote)": "'",
```

Comment:
Need the plurals to be consistent here. I'd have `(quotes | quote)` for the first one and `(single | thin) (quote | quotes)` for the second

Reply:
Actually we really need the language to be consistent here with regular caster. "quotes" refers to a pair and we don't have a word for a single quote. We need all of these in `punctuation.py` and ideally we would just rewrite `punctuation.py` to have this dictionary style and we'd just import it here.

Reply:
I'm going to have a go at refactoring `punctuation.py` to make it usable here

Reply:
You can try out the character dicts here https://github.com/dictation-toolbox/Caster/pull/548